### PR TITLE
fix(rush): install-run.js is not able to find its own .bin in PATH

### LINKED ISSUE #1917: [rush] Command rush can not be found when running through install-run-rush.js
**Is this a feature or a bug?**

- [ ] Feature
- [x] Bug

**Please describe the actual behavior.**

When running custom CLI command that uses `rush` underneath through `common/scripts/install-run-rush.js` it can not locate the `rush` command.

**If the issue is a bug, how can we reproduce it? Please provide detailed steps and include a GitHub branch if applicable. Your issue will get resolved faster if you can make it easy to investigate.**

Here is a reproducible minimal configuration:

```
{
  "$schema": "https://developer.microsoft.com/json-schemas/rush/v5/command-line.schema.json",
  "commands": [
    {
      "allowWarningsInSuccessfulBuild": false,
      "commandKind": "global",
      "description": "This command will clean everything and run the whole build pipeline",
      "name": "all",
      "safeForSimultaneousRushProcesses": true,
      "shellCommand": "rush clean && rush rebuild && rush test && rush lint",
      "summary": "Clean & Run the entire build pipeline (clean, build, test, lint) for all projects"
    }
  ]
}
```

The repository itself - https://github.com/ghaiklor/kittik/tree/3c2e8a212bfffede78c99bdb8dfdbe72f030ec3f
Link to the build - https://travis-ci.com/github/ghaiklor/kittik/jobs/344358951

Now, when running it like `node common/scripts/install-run-rush.js all` it can not find executable `rush`.

**What is the expected behavior?**

`rush` can be found and all the sub-tasks from Rush can be executed.

**If this is a bug, please provide the tool version, Node.js version, and OS.**

* **Tool:** Rush
* **Tool Version:** 5.24.3
* **Node Version:** v14.3.0
  * **Is this a LTS version?** pre-LTS
  * **Have you tested on a LTS version?** Yes
* **OS:** MacOS 10.15.5

### PR DESCRIPTION
**Pull request by ghaiklor**
Fixes #1917 

I'm not sure, though, how to write a regression test for this.

### CODE REVIEW

File: apps/rush-lib/src/scripts/install-run.ts [hunk 1]
```diff
@@ -437,13 +437,17 @@ export function installAndRun(
...
   const result: childProcess.SpawnSyncReturns<Buffer>  = childProcess.spawnSync(
     binPath,
     packageBinArgs,
     {
       stdio: 'inherit',
       cwd: process.cwd(),
-      env: process.env
+      env: {
+        ...process.env,
+        PATH: `${binFolderPath}:${process.env.PATH}`
```
Comment (octogonz) [inline apps/rush-lib/src/scripts/install-run.ts:449]: @ghaiklor Could you test this on both Windows and Mac/Linux?

IIRC the shell path separator may be different.  https://nodejs.org/api/path.html#path_path_delimiter

Comment (ghaiklor) [reply apps/rush-lib/src/scripts/install-run.ts]: Wonderful catch, thanks, updated. Regard to testing, only Mac here.

Comment (ghaiklor) [reply apps/rush-lib/src/scripts/install-run.ts]: Old `install-run-rush.js`:

```
Starting "rush all"
sh: rush: command not found
```

New `install-run-rush.js`:

```
Starting "rush all"

Rush Multi-Project Build Tool 5.24.3 - https://rushjs.io
Node.js version is 14.4.0 (pre-LTS)

Starting "rush clean"
# ...
```

File: common/changes/@microsoft/rush/master_2020-06-05-18-36.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/rush",
+      "comment": "Fixes an issue when install-run.js is not able to find its own .bin in PATH",
```
Comment (iclanton) [inline common/changes/@microsoft/rush/master_2020-06-05-18-36.json:5]:
```suggestion
      "comment": "Fix an issue where install-run.js is not able to find its own .bin in PATH",
```

**Comment by octogonz** [comment]
@ghaiklor Thanks for making this fix! It was published with Rush `5.24.4`

**Pull request merged**